src/player.ts

```diff
@@ -54,7 +54,7 @@ export default class Player extends EventEmitter {
       faction: this.faction,
       data: this.data,
       income: Reward.toString(Reward.merge([].concat(...this.events[Operator.Income].map(event => event.rewards))), true),
-      progress:  Object.assign({}, ...Object.values(FinalTile).map( track => ({ [track]: this.eventConditionCount(finalScorings[track])}))),
+      progress:  Object.assign({}, ...Object.values(FinalTile).map( track => ({ [track]: this.eventConditionCount(finalScorings[track])})), { structureValue: this.eventConditionCount(Condition.StructureValue) }, { structureFedValue: this.eventConditionCount(Condition.StructureFedValue) }),
```

Comment:
On the calculations: it think even all other factions can benefit from at least the structureValue. TBD

on the `Engine.fromData`. So the client can run locally the engine? But has to reprocess all moves? I've noticed that the response time towards the end of the game is increasing

Reply:
It doesn't have to reprocess all moves. 

The response time mainly comes from processing all the moves. And generating available commands. The client will be doing neither. It will just be converting the JSON data into a workable Engine.

Reply:
And yes, the client can run engine locally!

If you want, you can merge this PR for now, but when we use the Engine.fromData on the client side (including for final VP preview), this particular line should be reverted, in order to save a bit less data in db and make server load lighter.

We should also figure a way to send the map using less data than we are right now. The JSON data for the map is very verbose!